contracts/operators/StakeDAO/StakeDaoCurveStrategyOperator.sol

```diff
@@ -0,0 +1,379 @@
+        uint256 tokenBalanceBefore = outputToken.balanceOf(address(this));
+
+        _withdrawAndRemoveLiquidity128(
+            strategy,
+            amount,
+            ICurvePool(pool),
+            IERC20(lpToken),
+            poolCoinAmount,
+            address(outputToken)
+        );
+
+        (amounts, tokens) = CurveHelpers.getOutputAmounts(
+            IERC20(strategy),
+            strategyBalanceBefore,
+            amount,
+            outputToken,
+            tokenBalanceBefore,
+            minAmountOut
+        );
+    }
+
+    /// @notice Withdraw the LP token from the StakeDAO strategy,
+    ///         remove the liquidity from the Curve pool
+    ///         (using uint256 for the curvePool.remove_liquidity_one_coin
+    ///         coin index parameter) and receive one of the
+    ///         Curve pool token
+    /// @param strategy The StakeDAO strategy address to withdraw from
+    /// @param amount The amount to withdraw
+    /// @param outputToken Output token to receive
+    /// @param minAmountOut The minimum of output token expected
+    /// @return amounts Array of amounts :
+    ///         - [0] : The token received amount
+    ///         - [1] : The strategy token deposited amount
+    /// @return tokens Array of token addresses
+    ///         - [0] : The token received address
+    ///         - [1] : The strategy token deposited address
+    function withdraw256(
+        address strategy,
+        uint256 amount,
+        IERC20 outputToken,
+        uint256 minAmountOut
+    ) external payable returns (uint256[] memory amounts, address[] memory tokens) {
+        require(amount != 0, "SDCSO: INVALID_AMOUNT");
+
+        (address pool, uint96 poolCoinAmount, address lpToken) = operatorStorage.strategies(strategy);
+        require(pool != address(0), "SDCSO: INVALID_STRATEGY");
+
+        uint256 strategyBalanceBefore = IERC20(strategy).balanceOf(address(this));
+        uint256 tokenBalanceBefore = outputToken.balanceOf(address(this));
+
+        _withdrawAndRemoveLiquidity256(
+            strategy,
+            amount,
+            ICurvePoolNonETH(pool),
+            IERC20(lpToken),
+            poolCoinAmount,
+            address(outputToken)
+        );
+
+        (amounts, tokens) = CurveHelpers.getOutputAmounts(
+            IERC20(strategy),
+            strategyBalanceBefore,
+            amount,
+            outputToken,
+            tokenBalanceBefore,
+            minAmountOut
+        );
+    }
+
+    /// @dev  Add liquidity in a Curve pool with ETH and deposit
+    ///       the LP token in a StakeDAO strategy
+    /// @param strategy The StakeDAO strategy address to deposit into
+    /// @param pool The Curve pool to add liquitiy in
+    /// @param lpToken The Curve pool LP token
+    /// @param poolCoinAmount The number of token in the Curve pool
+    /// @param amount ETH amount to add in the Curve pool
+    function _addLiquidityAndDepositETH(
```

Comment:
`_addLiquidityAndDepositETH` and `_addLiquidityAndDeposit` are the same for Yearn and StakeDAO so they can be factorised for both Yearn and StakeDAO operators.

for `_withdrawAndRemoveLiquidity128` and `_withdrawAndRemoveLiquidity256`, Yearn and StakeDAO do not have the same interface for the shares withdraw because Yearn requires **3 parameters** for the call of the `withdraw` function:
- **sharesAmount**
- **recipient**
- **maxLoss**

But since the Yearn smart contract overloads the parameters with the following default values:
- **sharesAmount**: _MAX_UINT256_
- **recipient**: _msg.sender_
- **maxLoss**: _1 (0.01%)_

If we agree to always use the maxLoss (slippage) at 0.01% for Yearn, then we can refactor the `_withdrawAndRemoveLiquidity128` and `_withdrawAndRemoveLiquidity256` functions for Yearn and StakeDAO operators.

Reply:
👍